src/LibChorus/FileTypeHandlers/text/TextFileTypeHandler.cs

```diff
@@ -64,26 +66,36 @@ public static string GetRawMerge(string oursPath, string commonPath, string thei
 		{
 			//NB: surrounding with quotes didn't cut it to get past paths with spaces
 
-			return RunProcess("diff3/bin/diff3.exe", "-m " + LongToShortConverter.GetShortPath(oursPath) + " " +
-				LongToShortConverter.GetShortPath(commonPath) + " " +
-				LongToShortConverter.GetShortPath(theirPath));
+			return RunProcess(_diff3Exe, new string[] {"-m", LongToShortConverter.GetShortPath(oursPath),
+				LongToShortConverter.GetShortPath(commonPath),
+				LongToShortConverter.GetShortPath(theirPath) });
 		}
 
 		protected static string SurroundWithQuotes(string path)
 		{
 			return "\"" + path + "\"";
 		}
 
-		public static string RunProcess(string filePath, string arguments)
+		public static string RunProcess(string filePath, string[] arguments)
 		{
 			Process p = new Process();
+			ProcessStartInfo startInfo = new ProcessStartInfo();
 			p.StartInfo.UseShellExecute = false;
 			p.StartInfo.RedirectStandardError = true;
 			p.StartInfo.RedirectStandardOutput = true;
 
-			//NB: there is a post-build step in this project which copies the diff3 folder into the output
 			p.StartInfo.FileName = filePath;
-			p.StartInfo.Arguments = arguments;
+			// netstandard2.1 has p.StartInfo.ArgumentList that would solve all our problems, but we have to target netstandard2.0 so we have to do this ourselves
+			var quotedArgs = new List<string>(arguments.Length);
+			foreach (var arg in arguments)
+			{
+				if (arg.Contains(" ")) {
+					quotedArgs.Add("\"" + arg.Replace("\"", "\\\"") + "\"");
+				} else {
+					quotedArgs.Add(arg);
+				}
+			}
+			p.StartInfo.Arguments = string.Join(" ", quotedArgs);
```

Comment:
I don't love the idea of changing this, was there an issue with how it worked before?

Reply:
Yes, this code was not working, because the "test projéct" project has spaces in the filename and the quoting was incorrect. See https://github.com/sillsdev/chorus/actions/runs/11571890533/job/32210729290#step:7:4936 for an example of the errors that were happening: "ChorusMerge Error: Got error 2  diff3: extra operand '/tmp/hgmerge-ha1g3_90/one~other.txt'". That extra operand is because "test projéct", unquoted, was becoming two parameters, "test" and "projéct".

Furthermore, this failure was going *unnoticed* in the tests, because the test in question was expecting ChorusMerge to fail (due to an unresolvable merge conflict). But the diff3 code here was completely wrong, and so the ChorusMerge code these tests were *supposed* to test was never actually getting called.